Month2/week1/Month2-Project Week 1-Remembering the Bases/VideoGame/Steam.cpp

```diff
@@ -0,0 +1,317 @@
+#include "Steam.h"
+
+void FSteam::DebugApp() 
+{
+
+
+}
+
+FSteam::FSteam() 
+{
+	ActiveCommand = MenuCommand::MainMenu;
+}
+
+void FSteam::RunApp() 
+{
+	while (ActiveCommand != MenuCommand::Exit) 
+	{
+		switch (ActiveCommand)
+		{
+		case FSteam::MenuCommand::MainMenu:
+			OpenMainMenu();
+			break;
+		case FSteam::MenuCommand::AddGames:
+			OpenAddgamesMenu();
+			break;
+		case FSteam::MenuCommand::Categories:
+			OpenCategoriesMenu();
+			break;
+		case FSteam::MenuCommand::Exit:
+			break;
+		case FSteam::MenuCommand::VisualiceGames:
+			OpenVisualiceGamesMenu();
+		default:
+			break;
+		}
+	}
+}
+void FSteam::OpenMainMenu()
+{
+	system("CLS");
+	std::cout << "Welcome to Steam. What do You want to do?" << std::endl;
+	std::cout << "1.Add Games" << std::endl;
+	std::cout << "2.Browse Category" << std::endl;
+	std::cout << "3.Visualice Games" << std::endl;
+	std::cout << "Exit" << std::endl;
+	int OptionChoice;
+	std::cin >> OptionChoice;
+	if (!std::cin.good() || OptionChoice < 1 || OptionChoice > 4) 
+	{
+		std::cout << "Invalid Input, enter Your new choice" << std::endl;
+		std::cin.clear();
+		std::cin.ignore(1000, '\n');
+		std::cin >> OptionChoice;
+	}
+	ActiveCommand = (MenuCommand)OptionChoice;
+	system("CLS");
+}
+void FSteam::OpenAddgamesMenu()
+{
+	std::cin.clear();
+	std::cin.ignore(1000, '\n');
+	system("CLS");
+	std::cout << "Adding a VideoGame �Fantastic! Please Follow the process " << std::endl;
+	int Choice = 1;
+	do 
+	{
+		system("CLS");
+		std::cout << "Please enter the name of the Game" << std::endl;
+		std::string EntryGameName;
+		std::getline(std::cin, EntryGameName);
+		while (!std::cin.good() )
+		{
+			std::cin.good();
+			std::cin.ignore(1000, '\n');
+			std::cout << "Invalid input, please enter Your Game name again:" << std::endl;
+			std::getline(std::cin, EntryGameName);
+		}
+		std::cin.ignore(1000, '\n');
+		std::cout << "Please enter the name of the Studio " << std::endl;
+		std::string EntryStudioName;
+		std::getline(std::cin, EntryStudioName);
+		while (!std::cin.good() || EntryGameName=="")
+		{
+			std::cin.good();
+			std::cin.ignore(1000, '\n');
+			std::cout << "Invalid input, please enter Your Game name again:" << std::endl;
+			std::getline(std::cin, EntryStudioName);
+		}
+		std::cin.ignore(1000, '\n');
+		std::cout << "Please enter the day/Month/Year respectively " << std::endl;
+		std::cout << "Day: " << std::endl;
+		int EntryDay;
+		std::cin >> EntryDay;
+		if(EntryDay > 30 || EntryDay <= 0)
```

Comment:
Same as with the IsValid stuff for games, you're not allowing the user to enter 31st as a day of the publishing date